spec/models/user_spec.rb

```diff
@@ -22,3 +22,108 @@
     expect(user.next_slot_in_queue).to eq 1
   end
 end
+
+describe '#remove_from_queue!' do
+  let(:user) { Fabricate :user }
+
+  it 'deletes the input queue item, if it belongs to the current user' do
+    item_to_delete = Fabricate :queue_item, user: user
+    user.remove_from_queue!(item_to_delete)
+    expect(QueueItem.all.count).to eq 0
+  end
+
+  it 'does not delete the queue item, if it does not belong to current user' do
+    item_to_delete = Fabricate :queue_item
+    user.remove_from_queue!(item_to_delete)
+    expect(QueueItem.all.count).to eq 1
+  end
+end
+
+describe '#has_in_queue?' do
+  let(:user) { Fabricate :user }
+  let(:south_park) { Fabricate :video, title: 'South Park', tagline: 'hahahaha so so funny!' }
+
+  it "returns true if the given video is in the user's queue" do
+    Fabricate :queue_item, video: south_park, user: user
+    expect(user.has_in_queue? south_park).to be true
+  end
+
+  it "returns false if the given video is not in the user's queue" do
+    expect(user.has_in_queue? south_park).to be nil
```

Comment:
Your spec description says the method should return `false`, but your expectation checks for `nil`. Your description is correct, a question-mark method should return a boolean. You should update your expectation to check for `false`, then make sure it passes.

Be careful with stale (incorrect, out-dated) spec descriptions. Reading specs is one of the ways that new developers are onboarded to a project, so you want the descriptions to be right!